### PR TITLE
L1-292: `era_payout` unit tests

### PR DESCRIPTION
**Pull request by lesniak43**
# Description

Unit tests for `era_payout`.

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests

### CODE REVIEW

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -1589,4 +1590,155 @@ mod tests {
...
+                total_issuance: 50_000_000 * TOKEN,
+                era_duration_millis: MILLISECS_PER_DAY,
+            },
+            Outputs {
+                validators_payout: 123_118 * TOKEN + 920_000_000_000,
+                rest: 13_679 * TOKEN + 880_000_000_000,
+            },
+        );
+
+        // After 3 * horizon milliseconds the gap should be reduced by ~95%
```
Comment (DamianStraszak) [inline bin/runtime/src/lib.rs:1667]: Because (1/e)^3 = 0.05? Is this written somewhere that for 1 horizon we expect 1/e reduction?

**Review by DamianStraszak — APPROVED** [review]

File: bin/runtime/src/lib.rs (hunk 1 above)
Comment (lesniak43) [reply bin/runtime/src/lib.rs:1667]: Yup. And there's one check for a scenario in which era's length is equal to the horizon.

**Review by timorleph — CHANGES_REQUESTED** [review]
Having these as separate tests would be much clearer imo.

File: bin/runtime/src/lib.rs [hunk 2]
```diff
@@ -1589,4 +1590,155 @@ mod tests {
 
         assert!(lhs < rhs);
     }
+
+    #[test]
+    fn era_payout() {
```
[the remark this answers is not in the document]
Comment (lesniak43) [reply bin/runtime/src/lib.rs]: fixed

File: bin/runtime/src/lib.rs [hunk 3]
```diff
@@ -1589,4 +1590,155 @@ mod tests {
...
+        use sp_io::TestExternalities;
+
+        struct Inputs {
+            azero_cap: Balance,
+            horizon: u64,
+            // will be ignored
```
[the remark this answers is not in the document]
Comment (lesniak43) [reply bin/runtime/src/lib.rs]: fixed

**Review by timorleph — APPROVED** [review]

**Pull request merged**